### PR TITLE
Create ecosystem proto and implementation changes

### LINKED ISSUE #759: Create Ecosystem updates
The SDK logic for CreateEcosystem is now changed to:

When the submitted "name" or "email" is present, the client must send authenticated request, otherwise, send anonymous request.

Languages:
- [x] .NET https://github.com/trinsic-id/sdk/pull/756
- [x] CLI https://github.com/trinsic-id/sdk/pull/756
- [x] Go https://github.com/trinsic-id/sdk/pull/761
- [x] Java https://github.com/trinsic-id/sdk/pull/761
- [x] Dart https://github.com/trinsic-id/sdk/pull/761
- [x] Python https://github.com/trinsic-id/sdk/pull/761
- [x] Ruby https://github.com/trinsic-id/sdk/pull/761
- [x] Swift https://github.com/trinsic-id/sdk-swift/pull/20
- [x] Typescript https://github.com/trinsic-id/sdk/pull/756

### PR DESCRIPTION
**Pull request by tmarkovski**
- dotnet
- cli
- web (ts)

### CODE REVIEW

**Comment by fundthmcalculus** [comment]
I will fix the java issues in a PR, hold on a minute. Do we have an SDK change request in for this set of changes?

**Comment by tmarkovski** [comment]
You mean an issue? I don't think so. I'll make one.

File: dotnet/Trinsic/ProviderService.cs [hunk 1]
```diff
@@ -77,7 +77,9 @@ internal ProviderService(ITokenProvider tokenProvider, IOptions<ServiceOptions>
     public async Task<(Ecosystem ecosystem, string authToken)> CreateEcosystemAsync(CreateEcosystemRequest request) {
         request.Details ??= new();
 
-        var response = await Client.CreateEcosystemAsync(request);
+        var response = !string.IsNullOrWhiteSpace(request.Name) || !string.IsNullOrWhiteSpace(request.Details?.Email)
+            ? await Client.CreateEcosystemAsync(request, await BuildMetadataAsync(request))
+            : await Client.CreateEcosystemAsync(request);
```
Comment (fundthmcalculus) [inline dotnet/Trinsic/ProviderService.cs:82]: This change needs replicated.

**Comment by fundthmcalculus** [comment]
> You mean an issue? I don't think so. I'll make one.

https://github.com/trinsic-id/sdk/issues/759 <- thanks! Will do this as a set of PRs into this one, and we can merge all at once.

File: proto/services/account/v1/account.proto [hunk 2]
```diff
@@ -161,7 +161,7 @@ message LoginConfirmResponse {
     AccountProfile profile = 1;
 }
 
-// Authorize ecosystem to receive wallet events
+// Authorize ecosystem to receive wallet envents
```
Comment (fundthmcalculus) [inline proto/services/account/v1/account.proto]:
```suggestion
// Authorize ecosystem to receive wallet events
```

File: web/src/ProviderService.ts [hunk 3]
```diff
@@ -35,7 +35,13 @@ export class ProviderService extends ServiceBase {
...
     request: CreateEcosystemRequest
   ): Promise<CreateEcosystemResponse> {
-    return this.client.createEcosystem(request);
+    return request.name?.trim() || request.details?.email?.trim()
+      ? this.client.createEcosystem(request, {
+        metadata: await this.getMetadata(
+          CreateEcosystemRequest.encode(request).finish()
+        ),
+      })
+      : this.client.createEcosystem(request);
```
Comment (fundthmcalculus) [inline web/src/ProviderService.ts:44]: This needs replicated

**Review by fundthmcalculus — CHANGES_REQUESTED** [review]

**Comment by fundthmcalculus** [comment]
Made a quick typo fix in a proto comment, will merge new generated protos PR in. Other comments about "this needs replicated" are for my own reference, don't close, and nothing to fix.

**Review by fundthmcalculus — APPROVED** [review]

**Pull request merged**